File: _LLD/DesignPatterns/CreationalPatterns/FactoryMethod/PythonCode/FactoryMethodImpl.py
from abc import ABC, abstractmethod

# Super Class
class FactoryClass(ABC):
   def __init__(self) -> None:
      super().__init__()
      pass

   # This method will be overwritten/alter by the child class
   @abstractmethod
   def genericMethod():
      pass


# Child Class 1
class ChildOne(FactoryClass):
   def __init__(self) -> None:
      super().__init__()

   def genericMethod():
      print("this is method in child class1")

# Child Class 2
class ChildTwo(FactoryClass):
   def __init__(self) -> None:
      super().__init__()

   def genericMethod():
      print('this is method in child class2')

File: _LLD/DesignPatterns/CreationalPatterns/FactoryMethod/JavascriptCode/FactoryMethodImpl.js
// Super Class
class FactoryMethod {
  constructor() {}

  genericMethod() {}
}

// Child Class 1
class ChildOne extends FactoryMethod {
  constructor() {}

  genericMethod() {
    console.log('genericMethod in the child one');
  }
}

// Child Class 2
class ChildTwo extends FactoryMethod {
  constructor() {}

  genericMethod() {
    console.log('genericMethod in the child two');
  }
}
